model/src/main/java/com/chiralbehaviors/CoRE/Ruleform.java

```diff
@@ -104,6 +153,199 @@
+            if (field.getAnnotation(JoinColumn.class) == null) {
+                continue;
+            }
+            try {
+                field.setAccessible(true);
+                Ruleform value = (Ruleform) field.get(ruleform);
+                if (value != null && !ruleform.id.equals(value.id)) {
+                    traverser.accept(ruleform, field);
+                }
+            } catch (IllegalAccessException e) {
+                throw new IllegalStateException(String.format("IllegalAccess access foreign key field: %s",
+                                                              field),
+                                                e);
+            } catch (IllegalArgumentException e) {
+                throw new IllegalStateException(String.format("Illegal mapped value for field: %s",
+                                                              field),
+                                                e);
+            }
+        }
+    }
+
+    private static Ruleform asFrontier(Ruleform ruleform) {
+        Ruleform mappedValue = ruleform.clone();
+        for (Field field : getInheritedFields(ruleform.getClass())) {
+            if (field.getAnnotation(JoinColumn.class) != null
+                || field.getAnnotation(OneToMany.class) != null) {
+                try {
+                    field.setAccessible(true);
+                    field.set(mappedValue, null);
+                } catch (IllegalAccessException e) {
+                    throw new IllegalStateException(String.format("IllegalAccess access foreign key field: %s",
+                                                                  field),
+                                                    e);
+                } catch (IllegalArgumentException e) {
+                    throw new IllegalStateException(String.format("Illegal mapped value for field: %s",
+                                                                  field),
+                                                    e);
+                }
+            }
+        }
+        mappedValue.setNotes("Mapped frontier stand in");
+        return mappedValue;
+    }
+
+    private static Ruleform map(EntityManager em, Ruleform ruleform,
+                                Map<UUID, Ruleform> mapped) {
+        if (mapped.containsKey(ruleform.id)) {
+            return mapped.get(ruleform.id);
+        }
+
+        if (em.contains(ruleform)) {
+            mapped.put(ruleform.id, ruleform);
+            return ruleform;
+        }
+
+        Ruleform reference = find(em, ruleform);
+        BiConsumer<Ruleform, Field> slicer = (r, f) -> slice(r, f, em, mapped);
+        if (reference != null) {
+            Ruleform merged = em.merge(ruleform);
+            mapped.put(ruleform.id, em.merge(ruleform));
+            traverse(merged, slicer);
+        } else {
+            mapped.put(ruleform.id,
+                       em.getReference(ruleform.getClass(), ruleform.id));
```

Comment:
Using a reference here, instead of the actual object was literally brilliant and resolved a lot of circularity issues that drove me nuts.